plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -266,33 +268,41 @@ Job create(JobInfo jobInfo)
         return bigQuery.create(jobInfo);
     }
 
-    public void executeUpdate(QueryJobConfiguration job)
+    public void executeUpdate(ConnectorSession session, QueryJobConfiguration job)
     {
         log.debug("Execute query: %s", job.getQuery());
         try {
-            bigQuery.query(job);
+            bigQuery.query(setQueryLabel(session, job));
         }
         catch (InterruptedException e) {
             Thread.currentThread().interrupt();
             throw new BigQueryException(BaseHttpServiceException.UNKNOWN_CODE, format("Failed to run the query [%s]", job.getQuery()), e);
         }
     }
 
-    public TableResult query(String sql, boolean useQueryResultsCache, CreateDisposition createDisposition)
+    public TableResult query(ConnectorSession session, String sql)
     {
         log.debug("Execute query: %s", sql);
+        QueryJobConfiguration job = QueryJobConfiguration.newBuilder(sql)
+                .setUseQueryCache(isQueryResultsCacheEnabled(session))
+                .setCreateDisposition(createDisposition(session))
+                .build();
         try {
-            return bigQuery.query(QueryJobConfiguration.newBuilder(sql)
-                    .setUseQueryCache(useQueryResultsCache)
-                    .setCreateDisposition(createDisposition)
-                    .build());
+            return bigQuery.query(setQueryLabel(session, job));
         }
         catch (InterruptedException e) {
             Thread.currentThread().interrupt();
             throw new BigQueryException(BaseHttpServiceException.UNKNOWN_CODE, format("Failed to run the query [%s]", sql), e);
         }
     }
 
+    private QueryJobConfiguration setQueryLabel(ConnectorSession session, QueryJobConfiguration job)
+    {
+        return job.toBuilder()
+                .setLabels(ImmutableMap.of("trino_query_id", session.getQueryId()))
```

Comment:
I think the best would be to follow the same approach as we did with `query.comment-format`

Reply:
@kokosing would you like me to reuse existing code or copy it (it lives in base-jdbc for now)

Reply:
If possible to reuse then let's reuse.

Reply:
Done. Any last wishes? ;)